journal/src/main/java/io/zeebe/journal/JournalReader.java

```diff
@@ -91,6 +91,19 @@
    */
   long seekToAsqn(long asqn);
 
+  /**
+   * Seek to a record with the highest ASQN less than or equal to the given {@code asqn}, with the
+   * restriction that it seeks to a record with index less than or equal to the given
+   * indexUpperBound.
+   *
+   * <p>Otherwise, the semantics is similar to {@code seekToAsqn(asqn)}
+   *
+   * @param asqn
+   * @param indexUpperBound
```

Comment:
I think we can either add a description for the parameters or remove them if you think it's clear enough